### PR TITLE
Cache Package: return JError

### PR DESCRIPTION
**Pull request by AmyStephen**

### CODE REVIEW

File: libraries/joomla/cache/storage.php [hunk 1]
```diff
@@ -139,7 +141,7 @@ public static function getInstance($handler = null, $options = array())
 			}
 			else
 			{
-				return JError::raiseWarning(500, JText::sprintf('JLIB_CACHE_ERROR_CACHE_STORAGE_LOAD', $handler));
+				throw new RuntimeException('Cache Storage Class is not known.');
```
Comment (chdemko) [inline libraries/joomla/cache/storage.php]: Keep the sprintf

**Comment by AmyStephen** [comment]
Thanks @chdemko - should be ready now.

File: libraries/joomla/cache/storage.php [hunk 2]
```diff
@@ -114,7 +116,7 @@ public static function getInstance($handler = null, $options = array())
 			$handler = $conf->get('cache_handler');
 			if (empty($handler))
 			{
-				return JError::raiseWarning(500, JText::_('JLIB_CACHE_ERROR_CACHE_HANDLER_NOT_SET'));
+				throw new UnexpectedValueException(sprintf('Unable to load Cache Storage: %s', $handler));
```
Comment (realityking) [inline libraries/joomla/cache/storage.php]: UnexpectedValueException is great here but I don't think the message is accurate. This should only be thrown if no cache handler is set at all.

Comment (AmyStephen) [inline libraries/joomla/cache/storage.php]: Yup. You are right. Will change that.

**Comment by AmyStephen** [comment]
@realityking - done, thanks

**Comment by realityking** [comment]
Thanks!

**Pull request merged**